git_clases.rb

```diff
@@ -0,0 +1,107 @@
+require 'net/http'
+require 'json'
+require 'uri'
+# Esta clase es la encargada de crear un Gist, utilizando la API de GitHub.
+# @author Joel Alayon.
+# @since 0.9.24
+class Gist
+    # Esta variable almacena el estado del posteo.
+    # @return [String] estado del posteo.
+    # @example Posteo correcto:
+    #   response_status = "201"
+    # @example Error de conexión:
+    #   response_status = "400"
+    attr_reader :response_status
+    # Esta variable almacena la respuesta del usuario, con respecto a si quiere intentar otra vez.
+    # @note En caso de que haya un error en de conexión, se le preguntará al usuario si quiere intentar otra vez.
+    # @return [String] respuesta del usuario.
+    # @example Si el usuario quiere intentar otra vez:
+    #   try_again = "Si"
+    # @example Si el usuario NO quiere intentar otra vez:
+    #   try_again = "No"
+    attr_reader :try_again
+    # @author Joel Alayon.
+    # Recibe los parametros necesarios de instancia para poder instanciar el objeto.
+    # @note Los parametros del metodo de instancia se los pasa el usuario.
+    
+    def initialize(filename,description,state,content)
+        @filename = filename
+        @description = description
+        @state = state
+        @content = content
+    end
+
+    # @author Joel Alayon.
+    # Recibe los parametros necesarios para poder realizar la petición HTTP a la API.
+    # @note Los parametros de este metodo NO se los pasa el usuario.
+    # @see https://developer.github.com/v3/gists/ URL de la API.
+    def post(uri, request)
+        request.basic_auth("JoelAlayon123", "b545d404a388c4b900f2dfd520513c4f2daad292")
+        
+        request.body = JSON.dump({
+            "description" => @description,
+            "public" => @state,
+            "files" => {
+                @filename => {
+                    "content" => @content
+                }
+            }
+        })
+        
+        req_options = { use_ssl: uri.scheme == "https" }
+        
+        begin
+            response = Net::HTTP.start(uri.hostname, uri.port, req_options) do |http|
+                http.request(request)
+        end
+        
+        json = response.body
+        parsed = JSON.parse(json)
+        @response_status = "#{ response.code }"
```

Comment:
Why use string interpolation? You could do `@response_status = response.code` and then compare with integers instead of strings
`if @response_status == 201`

Reply:
Also, have a look at this https://github.com/airbnb/ruby#no-spaces-string-interpolation